jaxrs/src/main/java/com/facebook/airlift/jaxrs/JaxrsBinder.java

```diff
@@ -47,4 +56,10 @@ public void bindInstance(Object instance)
     {
         resourceBinder.addBinding().toInstance(instance);
     }
+
+    public void bindWithRolesMapping(Class<?> implementation, Map<String, String> roleMap)
```

Comment:
It is a little weird that `bind` is overloaded for `TypeLiteral<?>`, `Key<?>`, `instance` but the `bindWithRolesMapping ` isn't.

Ideally we should support all of the overloaded variants as well.

How about we change the structure to something like that

`jaxRsBinder.bind(ResourceImplementation.class).withRolesMap(roleMap)`

So basically we should make the `bind` methods to return and instance of `ResourceBinding` that is implemented something like

```
class ResourceBinding {
MapBinder<Class<?>, Map<String, String>> roleMapBinder;
...
public void withRolesMap(Map<String, String> roleMap){
roleMapBinder.addBinding(implementation).toInstance(roleMap);
}
}
```

Reply:
Since the mapping is based on class, I've made the change for `Class<?>` and `instance`(all instances of a same class will share annotation values, it should be fine we use the instance's class as key).
I'm not sure if it works for `TypeLiteral<?>` and `Key<?>` so I kept them unchanged. They are not being used in Presto neither.

Reply:
That's actually a little weird that we allow resource to be bound to a concrete instance  or by the key, as the single JaxRS server cannot handle multiple resource that have overlapping URL's declarations. But anyhow, since we already have this mechanism let's leave it as is. Feel free to do not implement roles mappings for these weird bindings.